scripts/core/editor3/components/Editor3.jsx

```diff
@@ -20,146 +21,137 @@ import Toolbar from './toolbar';
  * @description Editor3 is a draft.js based editor that support customizable
  *  formatting (shortly more features will come ...)
  */
-export class Editor3 extends React.Component {
-    constructor(props) {
-        super(props);
 
-        const initialContentState = stateFromHTML(props.value);
+export const getInitialState = (spellcheckerService, value, onChangeValue, readOnly, showToolbar, editorFormat) => {
+    const initialContentState = stateFromHTML(value);
 
-        this.state = {
-            editorState: EditorState.createWithContent(
+    return {
+        editorState: EditorState.createWithContent(
                 initialContentState,
-                this.getDecorators()
-            )
-        };
+                getDecorators(spellcheckerService)
+        ),
+        spellcheckerState: {spellcheckerService: spellcheckerService},
```

Comment:
The point is that this object is called `state`. When you reference it you write `state.spellcheckerState` which is stuttering and redundant. That's why I suggest renaming it to `spellchecker`. That way, when you say `state.spellchecker` it makes more sense.

An analogy would be a folder called "People". Inside you would have files names "Adrian, John, etc", not files named "PeopleAdrian, PeopleJohn, etc" or "AdrianPerson, John Person, etc".